liiatools_pipeline/sensors/location_schedule.py

```diff
@@ -0,0 +1,205 @@
+from hashlib import sha1
+
+import fs.errors
+from dagster import (
+    RunRequest,
+    RunConfig,
+    schedule,
+    RunsFilter,
+    DagsterRunStatus,
+)
+from fs import open_fs
+from fs.walk import Walker
+from decouple import config as env_config
+
+from liiatools_pipeline.jobs.common_la import clean
+from liiatools_pipeline.jobs.common_org import reports
+from liiatools_pipeline.ops.common_config import CleanConfig
+from liiatools.common.checks import check_la
+
+
+def input_directory_walker(folder_location, context, dataset):
+    """
+    Walks through the specified directory and returns a dictionary
+    of files for each LA
+    """
+    walker = Walker()
+    dir_pointer = open_fs(folder_location)
+    directories = dir_pointer.listdir("/")
+    dir_contents = {}
+
+    for directory_str in directories:
+        directory = open_fs(f"{folder_location}/{directory_str}/{dataset}")
+        dir_contents[directory_str] = [
+            file.lstrip("/") for file in walker.files(directory)
+        ]
+
+        if not dir_contents[directory_str]:
+            context.log.info(f"No files in {folder_location} have been found")
+
+    return dir_contents
+
+
+def concat_directory_walker(folder_location, context, dataset):
+    walker = Walker()
+    try:
+        concat_folder = open_fs(f"{folder_location}/concatenated/{dataset}")
+        context.log.info(f"Opening folder location: {folder_location}")
+        context.log.info("Analysing folder contents")
+        dir_contents = [file.lstrip("/") for file in walker.files(concat_folder)]
+
+    except fs.errors.CreateFailed:
+        context.log.info(
+            f"Failed to open folder location: {folder_location}/concatenated/{dataset}"
+        )
+        dir_contents = None
+
+    return dir_contents
+
+
+def generate_run_key(folder_location, files):
+    """
+    Generate a hash based on the last modified timestamps of the files.
+    """
+    hash_object = sha1()
+
+    for file_path in files:
+        with open_fs(folder_location) as filesystem:
+            last_modified_time = filesystem.getinfo(
+                file_path, namespaces=["details"]
+            ).modified
+
+            hash_object.update(str(last_modified_time).encode())
+
+    return hash_object.hexdigest()
+
+
+def find_previous_matching_run(
+    run_records, run_key, la_path, dataset, key_op, key_folder
+):
+    previous_run_id = [
+        run.dagster_run.tags["dagster/run_key"]
```

Comment:
This is causing errors in my tests as the key doesn't exist when it is run using a schedule (which is how the live system will run)

Could you make sure the tag exists in a scheduled run as well?